### PR TITLE
Model name can now be accessed through model.name

### PR DESCRIPTION
**Pull request by jack-w-shaw**
This resolves https://github.com/juju/python-libjuju/issues/693

Use model.name instead of model.info.name throughout codebase (inc tests)

### QA Steps

This change is now implicitly checked in some of the integration tests. So:
```
tox -e integration -- tests/integration/test_model.py::test_explicit_loop_threaded
```
should be sufficient

### CODE REVIEW

File: tests/integration/test_controller.py [hunk 1]
```diff
@@ -164,9 +164,9 @@ async def test_get_model(event_loop):
         try:
             by_name = await controller.get_model(model_name)
             by_uuid = await controller.get_model(model_uuid)
-            assert by_name.info.name == model_name
+            assert by_name.name == model_name
             assert by_name.info.uuid == model_uuid
-            assert by_uuid.info.name == model_name
+            assert by_uuid.name == model_name
             assert by_uuid.info.uuid == model_uuid
```
Comment (jack-w-shaw) [inline tests/integration/test_controller.py:170]: Is this a sign the same should be done with model.info.uuid?

File: juju/model.py [hunk 2]
```diff
@@ -978,6 +978,15 @@ def charmhub(self):
...
         return self._charmhub
 
+    @property
+    def name(self):
+        """Return the name of this model
+
+        """
+        if self.is_connected():
+            return self._info.name
+        return ""
```
[the remark this answers is not in the document]
Comment (juanmanuel-tirado) [reply juju/model.py:988]: Do we need to check the connection? That's not the usual procedure in the other model properties.

[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply juju/model.py:988]: If a model is not connected, then self._info is None

However this is Python so I forgot I should probably use `.get("name", "")` instead

[the remark this answers is not in the document]
Comment (cderici) [reply juju/model.py:988]: > If a model is not connected, then self._info is None

For now, yes. But your code relies on an extra assumption for no reason, such that, if that assumption ever changes in the future, then this code will break unnecessarily. That's why I think we shouldn't check the connection for this, and directly check if `self._info is None` instead.

Also, I believe returning `""` is incorrect. We should either return the actual name, or raise an error. Pulling the name of a model without connecting it first is incorrect/erroneous use. Not to mention returning `""` will cause any code that pulls `Model.name` to check for the case `""`.

**Review by cderici — APPROVED** [review]
This looks great to me. Let's land this as it is, the extra `asyncio` output is something we'll need to revisit again, not this PR's fault.

**Pull request merged**